config/kernel-access-ok-type.m4

```diff
@@ -0,0 +1,23 @@
+dnl #
+dnl # Linux 5.0: access_ok() drops 'type' parameter:
+dnl #
+dnl # - access_ok(type, addr, size)
+dnl # + access_ok(addr, size)
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_ACCESS_OK_TYPE], [
+	AC_MSG_CHECKING([whether access_ok() has 'type' parameter])
+	ZFS_LINUX_TRY_COMPILE([
+		#include <linux/uaccess.h>
+	],[
+		const void __user *addr = (void *) 0xdeadbeef;
+		unsigned long size = 1;
+		int ret = access_ok(0, addr, size);
+
+		return ret;
```

Comment:
This `return` might become problematic for some versions `gcc` since it makes the following `return 0` added by the `ZFS_LINUX_TRY_COMPILE` macro unreachable.  That warning could make this fail for the wrong reasons.  How about something like this, since we only care if this code compiles cleanly.

```c
	const void __user __attribute__((unused)) *addr = (void *) 0xdeadbeef;
	unsigned long __attribute__((unused)) size = 1;
	int error __attribute__((unused)) = access_ok(0, addr, size);
```